Testnet/Airdrop/Airdrop/Airdrop.cs

```diff
@@ -0,0 +1,202 @@
+        private set => PersistentState.SetAddress(nameof(TokenContractAddress), value);
+    }
+
+    /// <summary>The total supply of the token that will be distributed during this airdrop.</summary>
+    public ulong TotalSupply
+    {
+        get => PersistentState.GetUInt64(nameof(TotalSupply));
+        private set => PersistentState.SetUInt64(nameof(TotalSupply), value);
+    }
+
+    /// <summary>
+    /// Used to automatically close registration based on a specified block number. If the EndBlock is 0, the
+    /// registration period for the airdrop must be manually ended by the owner using <see cref="CloseRegistration"/>
+    /// </summary>
+    public ulong EndBlock
+    {
+        get => PersistentState.GetUInt64(nameof(EndBlock));
+        private set => PersistentState.SetUInt64(nameof(EndBlock), value);
+    }
+
+    /// <summary>Address of the owner of this contract, used to authenticate some methods.</summary>
+    public Address Owner
+    {
+        get => PersistentState.GetAddress(nameof(Owner));
+        private set => PersistentState.SetAddress(nameof(Owner), value);
+    }
+
+    /// <summary>The total number of registrants for this airdrop.</summary>
+    public ulong NumberOfRegistrants
+    {
+        get => PersistentState.GetUInt64(nameof(NumberOfRegistrants));
+        private set => PersistentState.SetUInt64(nameof(NumberOfRegistrants), value);
+    }
+
+    /// <summary>Calculates and sets the amount to distribute to each registrant.</summary>
+    public ulong AmountToDistribute
+    {
+        get
+        {
+            ulong amount = PersistentState.GetUInt64(nameof(AmountToDistribute));
+            if (amount == 0 && RegistrationIsClosed)
+            {
+                amount = TotalSupply / NumberOfRegistrants;
+                AmountToDistribute = amount;
+            }
+
+            return amount;
+        }
+
+        private set => PersistentState.SetUInt64(nameof(AmountToDistribute), value);
+    }
+
+    /// <summary>Returns whether or not the registration period is closed.</summary>
+    public bool RegistrationIsClosed
+    {
+        get
+        {
+            bool isClosed = PersistentState.GetBool(nameof(RegistrationIsClosed));
+            if (!isClosed && EndBlock > 0)
+            {
+                isClosed = Block.Number > EndBlock;
+                RegistrationIsClosed = isClosed;
+            }
+
+            return isClosed;
+        }
+
+        private set => PersistentState.SetBool(nameof(RegistrationIsClosed), value);
+    }
+
+    /// <summary>Returns the status of any given address.</summary>
+    public Status GetAccountStatus(Address address)
+    {
+        return PersistentState.GetStruct<Status>($"Status:{address}");
+    }
+
+    /// <summary>Sets the status for a given address.</summary>
+    private void SetAccountStatus(Address address, Status status)
+    {
+        PersistentState.SetStruct($"Status:{address}", status);
```

Comment:
Even though an enum is a struct, serializing them in this way will not work correctly. The recommendation for enums would be to cast to a `uint` and use `SetUInt32` to store them.